support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/response/accesstoken/OAuth20DefaultTokenGenerator.java

```diff
@@ -221,9 +221,14 @@ private OAuth20DeviceToken getDeviceTokenFromTicketRegistry(final String deviceC
      * Update OAuth code.
      *
      * @param holder the holder
+     * @param accessToken the accessToken
      */
-    protected void updateOAuthCode(final AccessTokenRequestDataHolder holder) {
-        if (holder.getToken() instanceof OAuth20Code) {
+    protected void updateOAuthCode(final AccessTokenRequestDataHolder holder, final String accessToken) {
+        if (holder.getToken() instanceof OAuth20RefreshToken) {
```

Comment:
Perhaps we could *hide* this a bit, by adding a isCodeToken() or isRefreshToken() type of methods to the holder, and move the conditions to the holder itself.

Reply:
Done